CHANGELOG.md

```diff
@@ -48,6 +48,33 @@ Swift Next
     }
   }
   ```
+  
+  As a result, this could lead to code that currently compiles today to throw an error.
+  
+  ```swift
+  protocol Foo {
+    var someProperty: Int { get set }
+  }
+  
+  class Bar: Foo {
+    var someProperty = 0
+  }
+  
+  extension Foo where Self: Bar {
+    var anotherProperty1: Int {
+      get { return someProperty }
+      // 'someProperty' exists as a protocol requirement (implicitly mutating
+      // due to the lack of a class constraint), so the setter below needs to 
+      // be explicitly marked as 'mutating'.
+      set { someProperty = newValue } // Error
+    }
+    
+    var anotherProperty2: Int {
+      get { return someProperty }
+      mutating set { someProperty = newValue } // Okay
```

Comment:
Wait, so the default changes to `nonmutating` but you're still allowed to *write* `mutating`? That's confusing but also a very easy answer. Can you add this as a test case to make sure it works?

Reply:
There's already a test case (see the last one). Do you mean a test case where I have explicitly written `mutating`?

Reply:
Actually, I _assumed_ we can add `mutating` to the setter (by looking at the fix-it), but I didn't actually try it. 

The fix-it is actually bogus because adding `mutating` will not make the error go away. What actually needs to be done is marking the protocol requirement as `nonmutating` as well, so we need to either not offer the fix-it or tweak it. I suppose I can do that in a follow-up PR.

Reply:
I was thinking the workaround is "assign into a temporary var, even though the change won't be reflected elsewhere". We definitely shouldn't offer the fix-it if it doesn't work. Changing the requirement probably isn't acceptable.

Reply:
Yeah, I've removed the fix-it now for that case.